Sources/Basic/TemporaryFile.swift

```diff
@@ -0,0 +1,172 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+
+ This file provides API to create and use temporary files and directories.
+*/
+
+import libc
+import func POSIX.getenv
+import class Foundation.NSFileHandle
+import var Foundation.NSUTF8StringEncoding
+import func Utility.fputs // FIXME: we should move file related helper methods to Basic.
+
+public enum TempFileError: ErrorProtocol {
+    /// Could not create a unique temporary filename.
+    case couldNotCreateUniqueName
+    /// The last 6 characters of template were not XXXXXX.
+    case badTemplateName
+    /// Some error thrown defined by posix's open().
+    // FIXME: This should be factored out into a open error enum.
+    case other(Int32)
+    /// Error while decoding unicode.
+    case unicodeDecodingError // FIXME: Already present in Utility.Error, Remove when file is shifted to basic.
+}
+
+private extension TempFileError {
+    init(errno: Int32) {
+        switch errno {
+        case libc.EEXIST:
+            self = .couldNotCreateUniqueName
+        case libc.EINVAL:
+            self = .badTemplateName
+        default:
+            self = .other(errno)
+        }
+    }
+}
+
+/// This class is basically a wrapper over posix's mkstemps() function to creates disposable files.
+/// The file is deleted as soon as the object of this class is deallocated.
+public final class TemporaryFile {
+    /// If specified during init, the temporary file name begins with this prefix.
+    public let prefix: String
+
+    /// If specified during init, the temporary file name ends with this suffix.
+    public let suffix: String
```

Comment:
Not sure if there is strong usecase for them to be public but since they're `lets` I guess there is no harm in exposing them? or do you think it'll lead to unclean API?